sgkit/variables.py

```diff
@@ -0,0 +1,225 @@
+from dataclasses import dataclass
+from typing import Dict, Hashable, Mapping, Set, Union, overload
+
+import xarray as xr
+
+
+@dataclass(frozen=True)
+class Spec:
+    """Root type Spec"""
+
+    default_name: str
+
+
+@dataclass(frozen=True)
+class ArrayLikeSpec(Spec):
+    """ArrayLike type spec"""
+
+    kind: Union[None, str, Set[str]] = None
+    ndim: Union[None, int, Set[int]] = None
+
+
+call_genotype = ArrayLikeSpec("call_genotype", kind="i", ndim=3)
+"""
+Call genotype. Encoded as allele values (0 for the reference, 1 for
+the first allele, 2 for the second allele), or -1 to indicate a
+missing value.
+"""
+call_genotype_mask = ArrayLikeSpec("call_genotype_mask", kind="b", ndim=3)
+"""TODO"""
+variant_contig = ArrayLikeSpec("variant_contig", kind="i", ndim=1)
+"""The (index of the) contig for each variant."""
+variant_position = ArrayLikeSpec("variant_position", kind="i", ndim=1)
+"""The reference position of the variant."""
+variant_allele = ArrayLikeSpec("variant_allele", kind={"S", "O"}, ndim=2)
+"""The possible alleles for the variant."""
+sample_id = ArrayLikeSpec("sample_id", kind={"U", "O"}, ndim=1)
+"""The unique identifier of the sample."""
+call_genotype_phased = ArrayLikeSpec("call_genotype_phased", kind="b", ndim=2)
+"""
+A flag for each call indicating if it is phased or not. If omitted
+all calls are unphased.
+"""
+variant_id = ArrayLikeSpec("variant_id", kind="U", ndim=1)
+"""The unique identifier of the variant."""
+call_dosage = ArrayLikeSpec("call_dosage", kind="f", ndim=2)
+"""Dosages, encoded as floats, with NaN indicating a missing value."""
+call_dosage_mask = ArrayLikeSpec("call_dosage_mask", kind="b", ndim=2)
+"""TODO"""
+call_genotype_probability = ArrayLikeSpec("call_genotype_probability", kind="f", ndim=3)
+"""TODO"""
+call_genotype_probability_mask = ArrayLikeSpec(
+    "call_genotype_probability_mask", kind="b", ndim=3
+)
+"""TODO"""
+genotype_counts = ArrayLikeSpec("genotype_counts", ndim=2, kind="i")
+"""
+Genotype counts. Must correspond to an (`N`, 3) array where `N` is equal
```

Comment:
FYI, the 1st sentence appears on the front page of the variable, the rest appears on the variable specific page.